cobigen/cobigen-core-api/src/main/java/com/devonfw/cobigen/api/util/TemplatesJarUtil.java

```diff
@@ -95,6 +101,68 @@ public static String downloadLatestDevon4jTemplates(boolean isDownloadSource, Fi
         TemplatesJarConstants.DEVON4J_TEMPLATES_ARTIFACTID, "LATEST", isDownloadSource, templatesDirectory);
   }
 
+  /**
+   * Downloads multiple jar files defined by the maven coordinates. Only downloads if files are not present or adapted
+   * folder does not exist.
+   *
+   * @param templatesDirectory directory where the templates jar are located
+   * @param mavenCoordinates list with {@link MavenCoordinate} that will be loaded
+   */
+  public static void downloadTemplatesByMavenCoordinates(Path templatesDirectory,
+      List<MavenCoordinate> mavenCoordinates) {
+
+    if (mavenCoordinates == null || mavenCoordinates.isEmpty()) {
+      return;
+      // no templates specified
+    }
+
+    Set<MavenCoordinate> existingTemplates = new HashSet<>();
+    Path adapted = templatesDirectory.resolve(ConfigurationConstants.ADAPTED_FOLDER);
+    Path downloaded = templatesDirectory.resolve(ConfigurationConstants.DOWNLOADED_FOLDER);
+    // search for already available template-sets
+    for (MavenCoordinate mavenCoordinate : mavenCoordinates) {
+      if (Files.exists(adapted)) {
+        try {
+          if (Files.list(adapted)
+              .anyMatch(f -> (f.getFileName().toString().contains(mavenCoordinate.getArtifactId())))) {
+            existingTemplates.add(mavenCoordinate);
+          }
+        } catch (IOException e) {
+          LOG.warn("Failed to get all files and directories from adapted folder", e);
+        }
+      }
+      if (Files.exists(downloaded)) {
+        try {
+          if (Files.list(downloaded)
+              .anyMatch(f -> (f.getFileName().toString().contains(mavenCoordinate.getArtifactId())))) {
+            existingTemplates.add(mavenCoordinate);
+          }
+        } catch (IOException e) {
+          LOG.warn("Failed to get all files and directories from downloaded folder", e);
+        }
```

Comment:
Looks like code duplication twice above

Reply:
Yeah i am going through the adapted and the downloaded folder to search for templates. I will extract the check into a private function.